plugins/gradle/gradle.plugin.zsh

```diff
@@ -60,6 +60,35 @@ _gradle_does_task_list_need_generating () {
   [[ ! -f .gradletasknamecache ]] || [[ build.gradle -nt .gradletasknamecache ]]
 }
 
+##############
+# Parse the tasks from `gradle(w) tasks --all` into .gradletasknamecache
+# All lines in the output from gradle(w) that are between /^-+$/ and /^\s*$/
+# are considered to be tasks. If and when gradle adds support for listing tasks
+# for programmatic parsing, this method can be deprecated.
+##############
+_gradle_parse_tasks_to_gradletasknamecache () {
+  lines_might_be_tasks=false
+  task_name_buffer=""
+  while read -r line; do
+    if [[ $line =~ ^-+$ ]]; then
+      lines_might_be_tasks=true
+      # Empty buffer, because it contains items that are not tasks
+      task_name_buffer=""
+    elif [[ $line =~ ^\s*$ ]]; then
+      if [[ "$lines_might_be_tasks" = true ]]; then
+        # If a newline is found, send the buffer to .gradletasknamecache
+        while read -r task; do
+          echo $task | awk '/[a-zA-Z0-9:-]+/ {print $1}' >> .gradletasknamecache
```

Comment:
Do you mean
```diff
-gradle tasks --all | awk '/[a-zA-Z0-9:-]* - / {print $1}' > .gradletasknamecache
+_gradle_parse_tasks "$(./gradlew tasks --all)"
```
Otherwise I didn't get it to work.